### PR TITLE
V1 of bringing everything together

### PR DESCRIPTION
**Pull request by stuartlynn**
This PR has the starting point of bringing the different threads together

- Makes everything async.
- Maps the CLI data command into a DataRequest 
- Uses this DataRequest along with the Metadata Catalogue to get the locations of the metrics and geoms 
- Call the methods to get the geometries and metrics 
- Join them together and return as a DataFrame 

Work still to be done that is probably for future PR's and issues.

1. Align with the new metadata model 
2. Explore accessing the different metric files in parallel to speed up fetches.
3. Figure out how the survey is selected / inferred.
4. How do we align async with wasm?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -10,8 +16,31 @@ pub struct Popgetter {
...
+        let geom_file = data_request.geom_details(&self.metadata)?;
+
+        // Required because polars is blocking
+        let metrics = tokio::task::spawn_blocking(move || {
+            get_metrics(&metric_requests,None)
+        });
+
+        let geoms = get_geometries(&geom_file, None, None);
+
+        // try_from requires us to have the errors from all futures be the same.
```
Comment (stuartlynn) [inline src/lib.rs:38]: This should be try_join!

File: src/lib.rs [hunk 2]
```diff
@@ -10,8 +16,31 @@ pub struct Popgetter {
...
+            get_metrics(&metric_requests,None)
+        });
+
+        let geoms = get_geometries(&geom_file, None, None);
+
+        // try_from requires us to have the errors from all futures be the same. 
+        // We use anyhow to get it back properly
+        let (metrics,geoms) = try_join!(async move { metrics.await.map_err(anyhow::Error::from)}, geoms)?;
+        
+        let result =metrics?.left_join(&geoms,["GEO_ID"],["GEOID"])?;
```
Comment (stuartlynn) [inline src/lib.rs:42]: These are hard coded but probably shouldn't be. We should use the convention of using GEOID everywhere but for now the test dataset has that column as "GEO_ID". 

Once we have a proper dataset to test with we can fix this

Comment (yongrenjie) [reply src/lib.rs:42]: Obviously the string itself should be a constant somewhere, we can fix this later though.

**Comment by yongrenjie** [comment]
Okay, I'm going to try to put together a dataset that obeys the metadata schema asap and we can go from there.

**Pull request merged**